command/hook_count.go

```diff
@@ -90,6 +91,11 @@ func (h *CountHook) PostDiff(
 	h.Lock()
 	defer h.Unlock()
 
+	// We don't count enything for data sources
+	if strings.HasPrefix(n.Id, "data.") {
```

Comment:
Might this have the same problem as discussed in #6823? Not sure if `n.Id` here is a module-local name or one that might potentially begin with `module.foo` for data resources in nested modules.